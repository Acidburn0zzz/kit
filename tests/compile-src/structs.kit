struct MyStruct {
    public static var xyz = "hi";

    var a: Int = 1;
    var b: CString = "hello";
    var c = 5;
    var d: Bool;
    var e: Ptr[Float];
}

function main() {
    var f = 5_f32;
    var a = struct MyStruct {
        c: 6,
        d: true,
        e: f,
    };
    // autorefderef of struct pointer field
    var g: Float = a.e;
    printf("%.1f\n", g);

    printf("%s\n", MyStruct.xyz);
    printf("%s\n", a.b);

    var b: Ptr[MyStruct] = &a;
    var y = b.a;
}
