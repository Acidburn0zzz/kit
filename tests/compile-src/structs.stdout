5.0
hi
hello
